docs/dev/Lua API.rst

```diff
@@ -5766,82 +5766,129 @@ The names of the functions are also available as the keys of the
-        if true make impassable tiles impassable to liquids too
-    :consume:
-        how much machine power is needed to work.
-        Disables reactions if not supplied enough and ``needs_power==1``
-    :produce:
-        how much machine power is produced.
-    :needs_power:
-        if produced in network < consumed stop working, default true
-    :gears:
-        a table or ``{x=?,y=?}`` of connection points for machines.
-    :action:
-        a table of number (how much ticks to skip) and a function which
-        gets called on shop update
-    :animate:
-        a table of frames which can be a table of:
+* ``fixImpassible(workshop_type, enable)``
 
-        a. tables of 4 numbers ``{tile,fore,back,bright}`` OR
-        b. empty table (tile not modified) OR
-        c. ``{x=<number> y=<number> + 4 numbers like in first case}``,
-           this generates full frame useful for animations that change little (1-2 tiles)
+  Set workshop non walkable tiles to also block liquids (i.e. water and magma).
 
-    :canBeRoomSubset:
-        a flag if this building can be counted in room. 1 means it can, 0 means it can't and -1 default building behaviour
-    :auto_gears:
-        a flag that automatically fills up gears and animations. It looks over the building definition for gear icons and maps them.
+  :workshop_type:   custom workshop string id e.g. ``SOAPMAKER`` or numeric id
+  :enable:   boolean value to enable or disable this functionality
 
-    Animate table also might contain:
+* ``setMachineInfo(workshop_type, needs_power, power_consumed, power_produced, connection_points)``
 
-    :frameLength:
-        how many ticks does one frame take OR
-    :isMechanical:
-        a bool that says to try to match to mechanical system (i.e. how gears are turning)
+  Setup and enable machine-like functionality for the workshop. Note: due to implementation limitations
+  workshop only connects to other machines if the other machine is build later than this one.
 
-``getPower(building)`` returns two number - produced and consumed power if building can be modified and returns nothing otherwise
+  :workshop_type:   custom workshop string id e.g. ``SOAPMAKER`` or numeric id
+  :needs_power:   only function if it has sufficient power
+  :power_consumed:   building consumes this amount of power
+  :power_produced:   output this amount of power
+  :connection_points:   a table of ``{x=?,y=?}`` points that can connect to other machines
 
-``setPower(building,produced,consumed)`` sets current power production and consumption for a building.
+* ``setMachineInfoAuto(workshop_type, needs_power, power_consumed, power_produced, gear_tiles)``
+
+  Same as ``setMachineInfo`` but fills out the ``connection_points`` table from raws placing connection
+  points on tiles which have the gear tile. ``gear_tiles`` is an optional array of two tiles that are
+  counted as gears in the workshop ascii tile raws.
+
+* ``setAnimationInfo(workshop_type, frames, frame_skip)``
```

Comment:
optional parameters need to be surrounded in square brackets, e.g. `setAnimationInfo(workshop_type, frames[, frame_skip])`